posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -0,0 +1,654 @@
+                    dag[label] = ModelNode(label=label)
+
+                node = dag[label]
+
+                if (
+                    (index == label_index or end > index > start)
+                    and label not in posthog_tables
+                    and node.selected is False
+                ):
+                    node = dag[label] = node.as_selected(True)
+
+                if index > 0:
+                    parent_node = path[index - 1]
+                    node.parents.add(parent_node)
+
+                if index < len(path) - 1:
+                    child_node = path[index + 1]
+                    node.children.add(child_node)
+
+    return dag
+
+
+async def get_posthog_tables(team_id: int) -> list[str]:
+    team = await database_sync_to_async(Team.objects.get)(id=team_id)
+    hogql_db = await database_sync_to_async(create_hogql_database)(team_id=team_id, team_arg=team)
+    posthog_tables = hogql_db.get_posthog_tables()
+    return posthog_tables
+
+
+@dataclasses.dataclass
+class StartRunActivityInputs:
+    dag: DAG
+    run_at: str
+    team_id: int
+
+
+@temporalio.activity.defn
+async def start_run_activity(inputs: StartRunActivityInputs) -> None:
+    """Activity that starts a run by updating statuses of associated models."""
+    run_at = dt.datetime.fromisoformat(inputs.run_at)
+
+    async with asyncio.TaskGroup() as tg:
+        for label in inputs.dag.keys():
+            tg.create_task(
+                update_saved_query_status(label, DataWarehouseSavedQuery.Status.RUNNING, run_at, inputs.team_id)
+            )
+
+
+@dataclasses.dataclass
+class FinishRunActivityInputs:
+    results: Results
+    run_at: str
+    team_id: int
+
+
+@temporalio.activity.defn
+async def finish_run_activity(inputs: FinishRunActivityInputs) -> None:
+    """Activity that finishes a run by updating statuses of associated models."""
+    run_at = dt.datetime.fromisoformat(inputs.run_at)
+
+    async with asyncio.TaskGroup() as tg:
+        for label in inputs.results.completed:
+            tg.create_task(
+                update_saved_query_status(label, DataWarehouseSavedQuery.Status.COMPLETED, run_at, inputs.team_id)
+            )
+
+        for label in itertools.chain(inputs.results.failed, inputs.results.ancestor_failed):
+            tg.create_task(
+                update_saved_query_status(label, DataWarehouseSavedQuery.Status.FAILED, run_at, inputs.team_id)
+            )
+
+
+async def update_saved_query_status(
+    label: str, status: DataWarehouseSavedQuery.Status, run_at: dt.datetime, team_id: int
+):
+    filter_params: dict[str, int | str | uuid.UUID] = {"team_id": team_id}
+
+    try:
+        model_id = uuid.UUID(label)
+        filter_params["id"] = model_id
+    except ValueError:
+        filter_params["name"] = label
+
+    saved_query = await database_sync_to_async(DataWarehouseSavedQuery.objects.filter(**filter_params).get)()
+    saved_query.last_run_at = run_at
+    saved_query.status = status
```

Comment:
Should this not be `last_run_status`?

Reply:
Some tests around this would be good

Reply:
There is now a test that runs the full workflow and asserts `status` and `last_run_at` are set properly. Also, `status` doesn't have a `last_run` prefix anymore.